app/src/main/java/com/epfl/drawyourpath/userProfileCreation/UserGoalsInitFragment.kt

```diff
@@ -78,18 +89,27 @@ class UserGoalsInitFragment : Fragment(R.layout.fragment_user_goals_init) {
             }
 
             if (test1 && test2 && test3) {
-                database.setUserGoals(username, distanceGoal.toDouble(), timeGoal.toDouble(), nunberOfPathGoal)
-
-                val previousActivity = activity
-                if (previousActivity != null) {
-                    val fragManagement = previousActivity.supportFragmentManager.beginTransaction()
-                    val dataToEndProfileCreationFrag: Bundle = Bundle()
-                    //data to transmit to the UserGoalsInitFragment(username)
-                    dataToEndProfileCreationFrag.putString("userName", username)
-                    val endProfileCreationFrag = EndProfileCreationFragment()
-                    endProfileCreationFrag.arguments = dataToEndProfileCreationFrag
-                    fragManagement.replace(R.id.userGoalInitFragment, endProfileCreationFrag)
-                        .commit()
+                var userLog = FirebaseAuth.getUser()
+                if(isTest){
+                    userLog = MockAuth.MOCK_USER
+                }
+                if(userLog == null) {
+                    throw java.lang.Error("Any user is logged to the app.")
+                }
+                val user = UserModel(userLog, firstname, surname, LocalDate.ofEpochDay(dateOfBirth), distanceGoal.toDouble(),
+                    timeGoal.toDouble(), nunberOfPathGoal, database)
+                database.initUserProfile(user).thenAccept{
+                    val previousActivity = activity
+                    if (previousActivity != null) {
+                        val fragManagement = previousActivity.supportFragmentManager.beginTransaction()
```

Comment:
The code to switch fragment is duplicated along all profile creation fragments. There should be a super class for all profile creation fragments that handles fragment transition.

Reply:
The problem is that the arguments transmitted between the fragment are not the same along the profile creation, so it will be difficult to create a commun function.

Reply:
You could allow the bundle as an argument to pass any data:
```kotlin
fun switchFragment<T>(fragId: Int, bundle: Bundle = Bundle()) {
    val fragManagement activity.supportFragmentManager.beginTransaction()
    fragManagement.replace(fragId, T()).commit()
}
```